### PR TITLE
Update several packages to the latest version and fix the subsequent lint violations

### PR DESCRIPTION
**Pull request by ar27111994**
# Description

This PR updates several packages to the latest version and fixes the subsequent lint violations introduced in forwarding and test helper paths, so the branch passes the project lint gate cleanly.

Summary of changes:
- Bump the dependencies group with 9 updates.
- Removed a useless assignment in the forwarding payload-size check.
- Preserved original caught errors by attaching cause metadata when rethrowing in test helpers.
- Removed an unused eslint-disable directive in unit tests.

Issue fixed:
- Resolves current lint failures on this branch (no-useless-assignment, preserve-caught-error, and unused eslint-disable warning).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

If you selected "Breaking change", please describe the impact and migration path below:

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have reviewed my changes for security implications

## Summary by CodeRabbit

* **Chores**
  * Bumped dev tooling and TypeScript; raised Node.js minimum to >=20.19.0.

* **Bug Fixes**
  * More robust webhook payload size determination to reliably enforce payload limits.

* **Tests**
  * Preserve original error causes in test diagnostics and harness failures.
  * Removed an unnecessary lint suppression.
  * Reorganized and expanded forwarding tests to cover Content-Length edge cases, retries, aborts, circuit-breaker behavior, header handling, logging, and security validations.

### CODE REVIEW

**Comment by ar27111994** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/ar27111994/webhook-debugger-logger/pull/55#pullrequestreview-4119863606)

File: src/services/ForwardingService.js [hunk 1]
```diff
@@ -268,20 +268,22 @@
...
-      bodySize = parseInt(String(req.headers[HTTP_HEADERS.CONTENT_LENGTH]), 10);
+    const parsedContentLength = parseInt(
+      String(req.headers[HTTP_HEADERS.CONTENT_LENGTH] ?? ""),
+      10,
+    );
+    let bodySize;
+    if (Number.isFinite(parsedContentLength)) {
+      bodySize = parsedContentLength;
+    } else if (Buffer.isBuffer(req.body)) {
+      bodySize = req.body.length;
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/services/ForwardingService.js:302]: False Positive: The `JSON.Stringify` already covers this.
